tests/ha/single_pod_restart/test_data_pod_restart_api.py

```diff
@@ -1012,3 +1015,210 @@ def test_copy_object_during_pod_restart(self):
+        etag = list(latest_vetag.values())[0]
+        if resp[1]["VersionId"] != latest_v:
+            assert_utils.assert_true(False, "Get Object without specifying VersionID does not "
+                                            f"match with latest {latest_v} {resp[1]}")
+        if resp[1]["ETag"] != etag:
+            assert_utils.assert_true(False, "Etag without specifying VersionID does not match with "
+                                            f"etag {etag} {resp[1]}")
+        LOGGER.info("Step 6: Got versions of %s with & without specifying VersionID & verified "
+                    "etags for %s", self.object_name, self.bucket_name)
+
+        new_bucket = self.bucket_name
+        download_path = self.multipart_obj_path
+        if CMN_CFG["dtm0_disabled"]:
+            download_path = os.path.join(self.test_dir_path, self.test_file + "_new")
+            self.extra_files.append(download_path)
+            new_bucket = f"ha-mp-bkt-{int(perf_counter_ns())}"
+            LOGGER.info("Step 7: Create new bucket and upload object %s of %s size. Enable "
+                        "versioning on %s.", self.object_name, self.f_size, new_bucket)
+            args = {'chk_null_version': True, 'is_unversioned': True, 'file_path': download_path,
+                    'enable_ver': True, 's3_ver': self.s3_ver}
+            resp = self.ha_api.crt_bkt_put_obj_enbl_ver(event, self.s3_test_obj, new_bucket,
+                                                        self.object_name, **args)
+            assert_utils.assert_true(resp[0], resp[1])
+            self.version_etag.update({new_bucket: []})
+            self.version_etag[new_bucket].extend(resp[1])
+            bucket_list.append(new_bucket)
+            LOGGER.info("Step 7: Created bucket and uploaded object %s of %s size. Enabled "
+                        "versioning on %s.", self.object_name, self.f_size, new_bucket)
+
+            LOGGER.info("Step 8: Upload same object %s after enabling versioning. List & verify "
+                        "the VersionID for the same for %s.", self.object_name, new_bucket)
+            args = {'file_path': download_path}
+            resp = self.ha_api.parallel_put_object(event, self.s3_test_obj, new_bucket,
+                                                   self.object_name, **args)
+            assert_utils.assert_true(resp[0], f"Upload Object failed {resp[1]}")
+            self.version_etag[new_bucket].extend(resp[1])
+            resp = self.ha_api.list_verify_version(self.s3_ver, new_bucket,
+                                                   self.version_etag[new_bucket])
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 8: Uploaded same object %s after enabling versioning. Listed & "
+                        "verified the VersionID for the same for %s.", self.object_name, new_bucket)
+
+        LOGGER.info("Step 9: Get versions of %s with & without specifying VersionID & verify "
```

Comment:
doesn't this also needs to go under DTM condition as for original bucket/object u verified this already in step 6, isn't it?